pkg/api/http/workflow.go

```diff
@@ -238,7 +238,19 @@ func (a *api) onTerminateWorkflowHandler() http.HandlerFunc {
 	return UniversalHTTPHandler(
 		a.universal.TerminateWorkflowBeta1,
 		UniversalHTTPHandlerOpts[*runtimev1pb.TerminateWorkflowRequest, *emptypb.Empty]{
-			InModifier:        workflowInModifier[*runtimev1pb.TerminateWorkflowRequest],
+			InModifier: func(r *http.Request, in *runtimev1pb.TerminateWorkflowRequest) (*runtimev1pb.TerminateWorkflowRequest, error) {
+				in.SetWorkflowComponent(chi.URLParam(r, workflowComponent))
+				in.SetInstanceId(chi.URLParam(r, instanceID))
+
+				// Default to true if not specified
+				in.Recursive = true
+				// Extract recursive option from query string
+				recursive := r.URL.Query().Get("recursive")
+				if recursive == "false" {
```

Comment:
why is the default `true`?

Reply:
That seems to be the intended behavior: https://github.com/microsoft/durabletask-go/pull/47#discussion_r1424793527
right @cgillum ?

Reply:
@shivamkm07 If the intended behaviour is to have a recursive terminate/purge, lets break the API ... But have the query string as `non-recursive`. It's is not the right behaviour to have a query string that is implicitly set to true.
cc @cgillum